keps/0019-package-api-versioning.md

```diff
@@ -0,0 +1,550 @@
+| T7   |  4.0.0-1.0.0 |       4.0.0 |             1.0.0 | v1               | KUDO API version change                    | Major bump               |
+
+##### Required and Suggested Changes
+
+1. Introduce the concept of the _operator revision_, which is an identifier that is re-set on every major/minor _app version_-based release. The "revision" naming is just to differentiate it from the monotonically increasing "version"
+2. Make the concept of the _operator version_ be the composition of the _app version_ and an _operator revision_. In this section, _package version_ and _operator version_ are the same thing (and will henceforth be called _version_), like in the current state of things with the singular version, and are also deterministic.
+
+   For example, in `2.3.0-1.0.0`:
+   - The _operator version_ (and by extension the _package version_, or just _version_) is `2.3.0-1.0.0`
+   - The _app version_ is `2.3.0`
+   - The _operator revision_ is `1.0.0`
+3. The _operator version_ wouldn't necessarily need to be explicitly set in operator or package metadata (`operator.yaml` and `index.yaml` respectively) given that the metadata contains fields for the _app version_ and the _operator revision_.
+
+It's important to note that **operator revision for different app versions are unrelated**. E.g., assuming two versions `2.3.0-1.0.0` and `3.0.0-1.0.0`, even though both have the same operator revision (`1.0.0`) they wouldn't necessarily have any commonality with regards to exclusively operator-related features and bug fixes. The operator revision progression is only meaningful within an app version's `major.minor` family, i.e. `2.3.x` or `3.0.x`.
+
+### CLI UX
+
+Independent of the version strategy, this will be the CLI UX
+
+| Concept                            | Flag          |
+| ------------------------           | ---------     |
+| Operator/package version (version) | --version     |
+| Application version                | --app-version |
+| Operator revision                  | --revision    |
+
+Assuming the following packages exist in the package registry
+
+for composite version:
+
+| Operator | Version     |
+| -------- | ----------- |
+| kafka    | 2.3.0-1.0.0 |
+| kafka    | 2.3.0-1.1.0 |
+| kafka    | 2.3.0-1.1.1 |
+| kafka    | 2.3.1-1.2.0 |
+| kafka    | 3.0.0-1.0.0 |
+| kafka    | 3.0.0-1.1.0 |
+| kafka    | 3.0.1-1.1.0 |
+| kafka    | 3.1.1-1.0.0 |
+
+for single version:
+                           
+| Operator | Version  |
+| -------- | -------- |
+| kafka    | 1.0.0    |
+| kafka    | 2.0.0    |
+| kafka    | 2.1.0    |
+| kafka    | 1.1.0    |
+| kafka    | 1.1.1    |
+| kafka    | 2.2.0    |
+| kafka    | 1.2.0    |
+| kafka    | 2.2.1    |
+| kafka    | 1.2.1    |
+
+**Install latest version of the Kafka operator**
+
+In the composite package version strategy we will need to specify how app-version is ordered. 
```

Comment:
In the beginning, it may be acceptable to simply err out and show a list of available appVersions for the user to specify, if multiple appVersions exist?